components/director/pkg/operation/update_operation_handler.go

```diff
@@ -0,0 +1,184 @@
+		return
+	}
+
+	tx, err := h.transact.Begin()
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("An error occurred while opening db transaction: %s", err.Error())
+		apperrors.WriteAppError(ctx, writer, apperrors.NewInternalError("Unable to establish connection with database"), http.StatusInternalServerError)
+		return
+	}
+	defer h.transact.RollbackUnlessCommitted(ctx, tx)
+
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	resourceUpdaterFunc := h.resourceUpdaterFuncs[operation.ResourceType]
+	opError, err := stringifiedJsonError(operation.Error)
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("An error occurred while marshalling operation error: %s", err.Error())
+		apperrors.WriteAppError(ctx, writer, apperrors.NewInternalError("Unable to marshal error"), http.StatusInternalServerError)
+		return
+	}
+
+	switch operation.OperationType {
+	case OperationTypeCreate:
+		fallthrough
+	case OperationTypeUpdate:
+		if err := resourceUpdaterFunc(ctx, operation.ResourceID, true, opError); err != nil {
+			log.C(ctx).WithError(err).Errorf("While updating resource %s with id %s", operation.ResourceType, operation.ResourceID)
+			apperrors.WriteAppError(ctx, writer, apperrors.NewInternalError("Unable to update resource %s with id %s", operation.ResourceType, operation.ResourceID), http.StatusInternalServerError)
+			return
+		}
+	case OperationTypeDelete:
+		resourceDeleterFunc := h.resourceDeleterFuncs[operation.ResourceType]
+		if operation.Error != "" {
+			if err := resourceUpdaterFunc(ctx, operation.ResourceID, true, opError); err != nil {
+				log.C(ctx).WithError(err).Errorf("While updating resource %s with id %s", operation.ResourceType, operation.ResourceID)
+				apperrors.WriteAppError(ctx, writer, apperrors.NewInternalError("Unable to update resource %s with id %s", operation.ResourceType, operation.ResourceID), http.StatusInternalServerError)
+				return
+			}
+		} else {
+			if err := resourceDeleterFunc(ctx, operation.ResourceID); err != nil {
+				log.C(ctx).WithError(err).Errorf("While deleting resource %s with id %s", operation.ResourceType, operation.ResourceID)
+				apperrors.WriteAppError(ctx, writer, apperrors.NewInternalError("Unable to delete resource %s with id %s", operation.ResourceType, operation.ResourceID), http.StatusInternalServerError)
+				return
+			}
+		}
+	}
+
+	if err := tx.Commit(); err != nil {
+		log.C(ctx).WithError(err).Errorf("An error occurred while closing database transaction: %s", err.Error())
+		apperrors.WriteAppError(ctx, writer, apperrors.NewInternalError("Unable to finalize database operation"), http.StatusInternalServerError)
+		return
+	}
+
+	writer.WriteHeader(http.StatusOK)
+}
+
+func operationRequestFromBody(ctx context.Context, request *http.Request) (*OperationRequest, *errResponse) {
+	bytes, err := ioutil.ReadAll(request.Body)
+	if err != nil {
+		return nil, &errResponse{apperrors.NewInternalError("Unable to read request body"), http.StatusInternalServerError}
```

Comment:
Probably http.StatusBadRequest is better in this case.

Reply:
In this case we try to read the request body which might not me a user (request) mistake